modules/polling/api/fetchPolls.ts

```diff
@@ -162,3 +173,205 @@ export async function getPolls(
     }
   };
 }
+
+export async function checkCachedPollsValidity(
+  network: SupportedNetworks
+): Promise<{ valid: boolean; hash?: string }> {
+  const isPollsHashValid: boolean | undefined = await cacheGet(isPollsHashValidCacheKey, network);
+
+  if (isPollsHashValid) {
+    return { valid: true };
+  }
+
+  const githubPollsHashRes = await fetch(POLLS_HASH_FILE_URL);
+  const githubPollsHashFile = await githubPollsHashRes.json();
+
+  const githubPollsHash: string | undefined = githubPollsHashFile.hash;
+  const cachedPollsHash: string | undefined = await cacheGet(pollsHashCacheKey, network);
+
+  if (githubPollsHash && cachedPollsHash && githubPollsHash === cachedPollsHash) {
+    cacheSet(isPollsHashValidCacheKey, 'true', network, THIRTY_MINUTES_IN_MS);
+    return { valid: true };
+  } else {
+    return { valid: false, hash: githubPollsHash };
+  }
+}
+
+export async function refetchPolls(
+  network: SupportedNetworks,
+  githubHash: string
+): Promise<{ pollList: PollListItem[]; allPolls: Poll[] }> {
+  const allPollsRes = await fetch(AGGREGATED_POLLS_FILE_URL);
+  const allPolls = await allPollsRes.json();
+
+  const pollList = allPolls.map(poll => {
+    const { pollId, startDate, endDate, slug, parameters, summary, title, options, tags } = poll;
+    return {
+      pollId,
+      startDate: new Date(startDate).toISOString(),
+      endDate: new Date(endDate).toISOString(),
+      slug,
+      type: parameters.inputFormat.type,
+      title,
+      summary,
+      options,
+      tags
+    };
+  });
+
+  cacheSet(isPollsHashValidCacheKey, 'true', network, THIRTY_MINUTES_IN_MS);
+  cacheSet(pollsHashCacheKey, githubHash, network, ONE_WEEK_IN_MS);
+  cacheSet(pollListCacheKey, JSON.stringify(pollList), network, ONE_WEEK_IN_MS);
+  allPolls.forEach((poll, i) => {
+    const prevPoll = allPolls[i - 1];
+    const nextPoll = allPolls[i + 1];
+
+    const pollWithLinks = {
+      ...poll,
+      ctx: {
+        prev: prevPoll ? { slug: prevPoll.slug } : null,
+        next: nextPoll ? { slug: nextPoll.slug } : null
+      }
+    };
+
+    const pollCacheKey = getIndividualPollCacheKey(pollWithLinks.pollId);
+    cacheSet(pollCacheKey, JSON.stringify(pollWithLinks), network, ONE_WEEK_IN_MS);
```

Comment:
okay, please add a comment here specifying what is different (poll detail includes more metadata like description, and we store separatedly in the cache to fetch it in the detail)

Reply:
Done